ls1intum/Artemis
# `Development`: Unify admin endpoints

### Checklist
#### General


#### Server
#### Client

### Motivation and Context


The idea of these changes is to prepare the endpoints for automated authorization tests. Additionally, it allows for autocompletion.

### Description

I moved all endpoints that are only accessible to Admins from `api/...` to `api/admin/...`, wrapped `@PreAuthorize("hasRole('ADMIN')")` in `@EnforceAdmin` and replaced all occurrences.
I also adapted the documentation and the pull request template.

### Steps for Testing

Look at the code and consider checking all affected endpoints (I also listed some functionality below). If there is no difference between endpoints in regard to the changes, testing them might not be necessary.
Keep in mind that this only affects permissions, not functionality. It's enough to check if you can see a resource/execute some action successfully.

Admin functionality:
- Create and delete courses
- Everything in the top right menu for admins
    - Upcoming exams & exercises
    - Manage users
    - Manage organisations
    - etc
- Trigger automatic assessment for text exercises

### Steps for Code Review
I made sure that all occurrences were replaced. Please double-check that I didn't accidentally replace too much and changed the permissions.
Also, ensure that I adapted all connections and didn't damage the interfaces.
If you have suggestions on how to improve the separation, let me know.

### Review Progress





#### Code Review
#### Manual Tests
